sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -238,10 +245,50 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
   private lazy val dateFormatter = DateFormatter()
   private lazy val timestampFormatter = TimestampFormatter.getFractionFormatter(zoneId)
 
+  private lazy val customDateFormatter: DateFormatter = format.map { f =>
```

Comment:
Is it possible to still use `dateFormatter` and `timestampFormatter` in `Cast` instead of custom formatters. For example, I would define them:
```Scala
private lazy val dateFormatter = format.map(DateFormatter(_)).getOrElse(DateFormatter())
private lazy val timestampFormatter = format
  .map(TimestampFormatter(_, zoneId))
  .getOrElse(TimestampFormatter.getFractionFormatter(zoneId))
```
Are there any downsides of that?

Reply:
Good idea, indeed.

Reply:
fixed thanks